core/extraction/towns_fund_round_three.py

```diff
@@ -288,9 +288,7 @@ def extract_project(df_project: pd.DataFrame, project_lookup: dict, programme_id
         lambda row: row[single_postcode] if row["Single or Multiple Locations"] == "Single" else row[multiple_postcode],
         axis=1,
     )
-    df_project["Postcodes"] = df_project.apply(
-        lambda row: ",".join(extract_postcodes(row["Locations"])), axis=1
-    )  # TODO: keep as a list and store as array of strings in db (not supported by SQLite)
+    df_project["Postcodes"] = df_project.apply(lambda row: ",".join(extract_postcodes(row["Locations"])), axis=1)
```

Comment:
I think this line, and the corresponding sub-function (split_string_by_delimiter) can all be replaced with something much more simple like 

`df_project["Postcodes"] = [extract_postcodes(x) for x in df_project["Locations"]]`

And then update extract_postcodes function to return None (or np.nan) type if no postcodes found. This then removes this double-handling of carting array to a string, then invoking another function per-line to iteratively convert back to array. It just does it in one go, without using apply twice.

Obvs, this will require some changes to everything else that extract_potcodes uses (I think some older extraction rounds also use it), but I think they are sensible, as otherwise we will have different extraction behaviour b/w rounds.